test/unit_test.cpp

```diff
@@ -0,0 +1,14 @@
+#define CATCH_CONFIG_MAIN
+
+#include <catch2/catch.hpp>
+
+unsigned int Factorial( unsigned int number ) {
```

Comment:
Since this is constructed as a stand-alone test, I'm not sure how one integrates a unit test for something in libnrniv.so.
Ie. a simple example of a test body for the interpreter add function in nrn/src/oc/code.c would be
hoc_pushx(4.0);
hoc_pushx(5.0);
hoc_add();
REQUIRE( hoc_xpop() == 9.0);

Reply:
I think above is good example!  Omar can replace the `Factorial` example with the above.